### ISSUE TITLE
zig build-lib doesn't emit memory.copy instruction on wasm target

### ISSUE THREAD

**Issue by rdunnington** [ziglang/zig#17278]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Behavior

Building this C code with `zig build-lib` or a `build.zig` results in a binary without `memory.copy` and `memory.fill` instructions replaced with calls to memcpy/memset, which leads to an infinite recursion. The issue is not present with `zig cc`.

```c
typedef unsigned int size_t;

void* memset(void* b, int c, size_t n)
{
    return (__builtin_memset(b, c, n));
}

void* memcpy(void* restrict dst, const void* restrict src, size_t n)
{
    return (__builtin_memcpy(dst, src, n));
}

char g_buffer[1024];

__attribute__((visibility("default"))) int set_stuff(void)
{
    memset(g_buffer, 0xFF, sizeof(g_buffer));
    return 0;
}

__attribute__((visibility("default"))) int copy_stuff(void* data, size_t len)
{
    memcpy(g_buffer, data, len);
    return 0;
}
```

The working `zig cc`: `zig cc --target=wasm32-wasi --no-standard-libraries -fno-builtin -mbulk-memory -mexec-model=reactor -Wl,--export-dynamic -o ./module.wasm main.c`

The `zig build-lib` that repros the problem: `zig build-lib -cflags --target=wasm32-freestanding --no-standard-libraries -fno-builtin -Xlinker --no-entry -Xlinker --export-dynamic -g  -bulk-memory -- main.c -target wasm32-freestanding -dynamic`

Note that the `zig cc` target specifies `wasm32-wasi` but the `zig build-lib` specifies `wasm32-freestanding`. I couldn't get `zig cc` to build for the freestanding target - it complained no entrypoint was present.

Also I would have linked to a godbolt, but it doesn't have the ability to use `zig build-lib` for C sources. :(

### Expected Behavior

Building C source with `zig build-lib` and `-bulk-memory` should emit `memory.copy` and `memory.fill` instructions.

**Comment by Luukdegram** [comment]
Your `zig build-lib` command doesn't enable the `bulk-memory` feature. Also the command doesn't export any function so an empty binary will be built. If you replace your command with the following it, it works correctly: `zig build-lib -cflags --target=wasm32-freestanding --no-standard-libraries -fno-builtin -Xlinker --no-entry -Xlinker --export-dynamic -g  -bulk-memory -- main.c -target wasm32-freestanding -dynamic -mcpu=generic+bulk_memory -rdynamic`

ps: I believe `-cflags` only works on positional c source files that are not the main entry, so all of the flags you passed to it with `build-lib` are ignored.

**Comment by rdunnington** [comment]
Ahh ok that makes sense, thank you! I did try with the bulk-memory feature but it was in the cflags so I guess it got ignored. This also gave me enough info to figure out how to do it via the build.zig route too - I needed to add the `addFeature` line to ensure the feature got added to the target cpu:
```zig
    var wasm_target = std.zig.CrossTarget{
        .cpu_arch = .wasm32,
        .os_tag = .freestanding,
    };

    wasm_target.cpu_features_add.addFeature(@intFromEnum(std.Target.wasm.Feature.bulk_memory));
```
Side note, I've tried looking for help on the zig cmdline features but it seems documentation is pretty sparse. Thanks again for the help!
